test/regression.carp

```diff
@@ -51,6 +51,35 @@
     (Maybe.Nothing) true
     _ (unreachable "test unreachable")))
 
+;; defining function with def (not defn)
+(defn duplicate-arg [f]
+  (fn [x] (f x x)))
+
+(def double (duplicate-arg +))
+
+(defn defining-function-with-def []
+  (double 3))
+
+;; using special symbol as binder name
```

Comment:
Is the test here just defining it? If so, maybe a mention of the issue number would be good!